t/bulk.t

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use FindBin qw( $Bin );
+use lib qw( t/lib );
+use Test::More;
+use Sympa::Tools::File;
+use Sympa::Message;
+use ok 'Sympa::Bulk';
+use English;
+
+# during a test, syslog would be helpful as a note
+# so *Sympa::Log::syslog is diverted
+#
+# by default, syslog remains silent
+
+{ no warnings;
+    *Sympa::Log::syslog = 1 #  $ENV{SYMPA_TEST_SILENT_SYSLOG}
+    ? sub {}
+    : sub {
+        shift; # $self isn't expected anymore
+        my ( $level, $fmt, @params ) = @_;
+        note sprintf "syslog $level: $fmt", @params; }
+    ;
+}
+
+# this is the basic (ideally the only one keys required)
+# config to make this test work
+
+%Conf::Conf = (
+    # Sympa::Bulk::store needs
+    sympa_packet_priority => 5,
+    sympa_priority        => 1,
+    email                 => 'sympa',
+    domain                => 'lists.example.com',
+    sender_headers        => 'Resent-From,From,Return-Path',
+    list_check_suffixes   => 'list_check_suffixes',
+    umask                 => '027',
+);
+
+# we need an absolute path there because in Sympa/Bulk.pm:109
+#
+#    $self->{_metadatas} = [
+#        sort grep {
+#                    !/,lock/
+#                and !m{(?:\A|/)(?:\.|T\.|BAD-)}
+#                and -f ($self->{pct_directory} . '/' . $_)
+#        } glob $self->{_glob_pattern}
+#    ];
+#
+# as we are already in $self->{pct_directory}, -f will always fail with a
+# relative path.
+#
+# i (eiro) don't know if it's a bug or a feature
+
+use Cwd;
+$Conf::Conf{queuebulk} =
+    Cwd::abs_path
+    Sympa::Constants::SPOOLDIR."/bulk";
+
```

Comment:
Directorie paths defined in Sympa::Constants may be assumed to be absolute path.  (Except in the test cases explicitly assuming definition of relative paths,) SPOOLDIR should have been defined as absolute path in Sympa::Constants, and fix up here seems inappropriate.

Reply:
i wrote it this way to keep a trace on the (i think undocumented) fact that those directories *must* be absolute. i'll check the doc on it

Reply:
At any rate, the "use Cwd" belongs to the top of the file, there is no point in:

- break it up in multiple lines
- skip the parenthesis
- use concat operator without whitespace